src/Orleans/Serialization/OrleansJsonSerializer.cs

```diff
@@ -13,15 +8,16 @@ namespace Orleans.Serialization
 {
     internal class OrleansJsonSerializer : IExternalSerializer
     {
-        public static JsonSerializerSettings settings;
+        internal static JsonSerializerSettings SerializerSettings { get; private set; }
+
         private TraceLogger logger;
 
         static OrleansJsonSerializer()
         {
-            settings = new JsonSerializerSettings
+            SerializerSettings = new JsonSerializerSettings
             {
                 TypeNameHandling = TypeNameHandling.All,
-                PreserveReferencesHandling = PreserveReferencesHandling.Objects,
+                PreserveReferencesHandling = PreserveReferencesHandling.All,
```

Comment:
This is the only change in settings I can see. Why do we need it as there seems to be no intent in changing the settings here? Or did I misunderstand the intent?